### PR TITLE
Populate MT report with stats from d4 files

### LINKED ISSUE #4836: Create MT reports using chanjo2
for the cases that have d4 files associated, that is

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Closes #4836

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Go to case [safeguinea](https://scout-stage.scilifelab.se/cust000/F0006358-mip7) and download the MT report. Make sure it contains the MT vs Autosomes stats

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR

### CODE REVIEW

**Comment by northwestwitch** [comment]
Hi @jemten , I'm working on the MT report using d4 files. I'm testing with [a stage case with associated d4 files](https://scout-stage.scilifelab.se/cust000/F0006358-mip7) but on the MT chrom I'm getting always coverage 0. Question: does the d4 contain MT at all? Are the MT stats on another file?

**Comment by northwestwitch** [comment]
> Hi @jemten , I'm working on the MT report using d4 files. I'm testing with [a stage case with associated d4 files](https://scout-stage.scilifelab.se/cust000/F0006358-mip7) but on the MT chrom I'm getting always coverage 0. Question: does the d4 contain MT at all? Are the MT stats on another file?
> 
> @ramprasadn @rannick

The d4 file in question:
/home/proj/stage/housekeeper-bundles/safeguinea/2023-06-26/ADM997A3_lanes_12_sorted_md.d4

Do we have a more recent example of d4 which I can use to test with?

**Comment by northwestwitch** [comment]
This is ready for review now. I've substituted the d4 file with one that has all chromosomes and it's testable with [safeguinea](https://scout-stage.scilifelab.se/cust000/F0006358-mip7)

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ About changelog [here](https://keepachangelog.com/en/1.0.0/)
 - Button on SMN CN page to search variants within SMN1 and SMN2 genes
 - Options for selectively updating OMICS variants (fraser, outrider) on a case
 - Log users' activity to file by specifying `USERS_ACTIVITY_LOG_PATH` parameter in app config
+- `Mean MT coverage`, `Mean chrom 14 coverage` and `Estimated mtDNA copy number` on MT coverage file will be obtained from chanjo2 if d4 files are associated to cases's individuals/samples
```
Comment (dnil) [inline CHANGELOG.md:12]:
```suggestion
- `Mean MT coverage`, `Mean chrom 14 coverage` and `Estimated mtDNA copy number` on MT coverage file from chanjo2 if available
```

File: scout/server/extensions/__init__.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
 
 from .beacon_extension import Beacon
 from .bionano_extension import BioNanoAccessAPI
+from .chanjo2_extension import Chanjo2Client
```
Comment (dnil) [inline scout/server/extensions/__init__.py:13]: 👍

File: scout/server/blueprints/cases/controllers.py [hunk 3]
```diff
@@ -725,9 +726,15 @@ def mt_excel_files(store, case_obj, temp_excel_dir):
...
     samples = case_obj.get("individuals")
     coverage_stats = None
-    # if chanjo connection is established, include MT vs AUTOSOME coverage stats
-    if current_app.config.get("chanjo_report"):
-        coverage_stats = chanjo_report.mt_coverage_stats(individuals=samples)
+
+    case_has_chanjo_coverage(case_obj)
+    case_has_chanjo2_coverage(case_obj)
+
+    # Check if coverage and MY copy number stats are available via chanjo2 or chanjo
```
Comment (dnil) [inline scout/server/blueprints/cases/controllers.py:733]:
```suggestion
    # Check if coverage and MT copy number stats are available via chanjo2 or chanjo
```

File: scout/server/extensions/chanjo2_extension.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+REF_CHROM = "14"
+MT_CHROM = "MT"
+LOG = logging.getLogger(__name__)
+
+
+class Chanjo2Client:
+    """Runs requests to chanjo2 and returns results in the expected format."""
+
+    def mt_coverage_stats(self, individuals: dict) -> Dict[str, dict]:
```
Comment (dnil) [inline scout/server/extensions/chanjo2_extension.py:15]: 👍

File: tests/server/extensions/test_chanjo2_extension.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+@responses.activate
+def test_chanjo2_mt_coverage_stats(case_obj):
+    """Test the function that sends requests to Chanjo2 to get MT vs autosomal coverage stats"""
+
+    # GIVEN a case with individuals containing d4 files
+    for ind in case_obj["individuals"]:
+        assert ind["d4_file"]
+
+    # GIVEN a patched chanjo2 instance
+    # GIVEN a mocked Beacon server dataset add endpoint
```
Comment (dnil) [inline tests/server/extensions/test_chanjo2_extension.py:18]:
```suggestion
    # GIVEN a mocked Chanjo2 server dataset add endpoint
```

File: tests/server/extensions/test_chanjo2_extension.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    url = f"{CHANJO2_BASE_URL}/coverage/d4/interval/"
+    responses.add(
```
Comment (dnil) [inline tests/server/extensions/test_chanjo2_extension.py:20]: 💯

**Review by dnil — APPROVED** [review]
Super! Really nice to see the chanjo2 extension on here! 🥇

**Pull request merged**